### PR TITLE
Update release documentation checks and templates

### PR DESCRIPTION
**Pull request by SergioEstevao**
To test:
 - Check if document and template is correct.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](docs/Release-notes.md) and have added them to `[RELEASE-NOTES.txt](RELEASE-NOTES.txt)` if necessary.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request.md [hunk 1]
```diff
@@ -28,11 +28,12 @@ Release for Gutenberg Mobile v1.XX.Y
...
 ## Release Submission Checklist
 
-- [ ] Release number was bumped
+- [ ] Release number was bumped in `package.json` and `gutenberg/packages/react-native-editor/package.json`
 - [ ] Aztec dependencies are pointing to a stable release
   - iOS: 'grep WordPressAztec-iOS RNTAztecView.podspec'
   - Android: 'grep aztecVersion react-native-aztec/android/build.gradle'
-- [ ] Gutenberg 'Podfile' and 'Podfile.lock' inside './ios/' are updated to the release number
-- [ ] Bundle package of the release is updated 
+- [ ] Aztec dependencies match in `RNTAztecView.podspec` and `gutemberg/packages/react-native-aztec/RNTAztecView.podspec`
```
Comment (mchowning) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request.md:35]: Minor typo: gutemberg -> gutenberg

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request.md [hunk 2]
```diff
@@ -28,11 +28,12 @@ Release for Gutenberg Mobile v1.XX.Y
 
 ## Release Submission Checklist
 
-- [ ] Release number was bumped
+- [ ] Release number was bumped in `package.json` and `gutenberg/packages/react-native-editor/package.json`
 - [ ] Aztec dependencies are pointing to a stable release
```
Comment (mchowning) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request.md:32]: It looks like the stable release check was dropped, was that intentional?

Comment (SergioEstevao) [reply .github/PULL_REQUEST_TEMPLATE/release_pull_request.md:32]: The check is still there, noticed that the line is not red. I think the minus for the list is making it look deleted.

Comment (mchowning) [reply .github/PULL_REQUEST_TEMPLATE/release_pull_request.md]: 🤦‍♂️ You're totally right!

**Pull request merged**